### PR TITLE
Debt/MET-5906 Tech Debt

### PR DESCRIPTION
**Pull request by JoanaCMS**

### CODE REVIEW

File: metis-common/metis-common-utils/src/main/java/eu/europeana/metis/utils/CompressedFileHandler.java
Comment (jeortizquan) [inline metis-common/metis-common-utils/src/main/java/eu/europeana/metis/utils/CompressedFileHandler.java:1]: Is something I forget to configure? I am reviewing with the sandbox and get this issue.
```
2024-04-02T14:26:36.565+02:00 ERROR eu.europeana.metis.sandbox.service.dataset.DatasetLogServiceImpl:52 [sandbox-1] - Exception occurred while sending records to execute: 'org.apache.commons.compress.archivers.zip.ZipArchiveEntry org.apache.commons.compress.archivers.zip.ZipArchiveInputStream.getNextEntry()'
java.lang.NoSuchMethodError: 'org.apache.commons.compress.archivers.zip.ZipArchiveEntry org.apache.commons.compress.archivers.zip.ZipArchiveInputStream.getNextEntry()'
	at eu.europeana.metis.utils.CompressedFileHandler.extractZipFile(CompressedFileHandler.java:128) ~[metis-common-utils-12-SNAPSHOT.jar:?]
	at eu.europeana.metis.utils.CompressedFileHandler.extractFile(CompressedFileHandler.java:61) ~[metis-common-utils-12-SNAPSHOT.jar:?]
	at eu.europeana.metis.harvesting.http.HttpHarvesterImpl.harvestRecords(HttpHarvesterImpl.java:118) ~[metis-harvesting-12-SNAPSHOT.jar:?]
	at eu.europeana.metis.harvesting.http.HttpHarvesterImpl.createTemporaryHttpHarvestIterator(HttpHarvesterImpl.java:103) ~[metis-harvesting-12-SNAPSHOT.jar:?]
	at eu.europeana.metis.sandbox.service.workflow.HarvestServiceImpl.harvestInputStreamIdentifiers(HarvestServiceImpl.java:221) ~[classes/:?]
	at eu.europeana.metis.sandbox.service.workflow.HarvestServiceImpl.harvest(HarvestServiceImpl.java:204) ~[classes/:?]
	at eu.europeana.metis.sandbox.service.workflow.HarvestPublishServiceImpl.lambda$runHarvestFileAsync$1(HarvestPublishServiceImpl.java:90) ~[classes/:?]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
```

**Review by jeortizquan — COMMENTED** [review]
Is a dependency that I need to update?

File: metis-authentication/metis-authentication-rest/src/main/java/eu/europeana/metis/authentication/rest/config/ApplicationConfiguration.java [hunk 1]
```diff
@@ -122,7 +124,8 @@ public SessionFactory getSessionFactory(HibernateConfigurationProperties hiberna
     try (Session dbSession = sessionFactory.openSession()) {
       performAction(dbSession, session -> {
         Transaction tx = session.beginTransaction();
-        session.createNativeQuery(createTablesSql).executeUpdate();
+        NativeQueryImpl<Void> nativeQuery = new NativeQueryImpl<>(createTablesSql, (SharedSessionContractImplementor) session);
```
Comment (stzanakis) [inline metis-authentication/metis-authentication-rest/src/main/java/eu/europeana/metis/authentication/rest/config/ApplicationConfiguration.java:127]: I think that won't work with the current file that is executed.
From what I remember this method does not accept an sql "script" which contain multiple sql commands.
Have you tested it?

File: metis-common/metis-common-utils/src/main/java/eu/europeana/metis/utils/CompressedFileHandler.java
Comment (JoanaCMS) [reply metis-common/metis-common-utils/src/main/java/eu/europeana/metis/utils/CompressedFileHandler.java:1]: Pull request in sandbox created, it should be merged after these changes are merged
https://github.com/europeana/metis-sandbox/pull/159

File: metis-authentication/metis-authentication-rest/src/main/java/eu/europeana/metis/authentication/rest/config/ApplicationConfiguration.java (hunk 1 above)
Comment (JoanaCMS) [reply metis-authentication/metis-authentication-rest/src/main/java/eu/europeana/metis/authentication/rest/config/ApplicationConfiguration.java:127]: I'm not able to test this, I'll just put it back to how it was before

**Review by jeortizquan — APPROVED** [review]

**Pull request merged**